src/Microsoft.ML.Recommender/MatrixFactorizationTrainer.cs

```diff
@@ -403,18 +393,22 @@ public MatrixFactorizationPredictionTransformer Fit(IDataView input)
             roles.Add(new KeyValuePair<RoleMappedSchema.ColumnRole, string>(RecommenderUtils.MatrixColumnIndexKind.Value, MatrixColumnIndexName));
             roles.Add(new KeyValuePair<RoleMappedSchema.ColumnRole, string>(RecommenderUtils.MatrixRowIndexKind.Value, MatrixRowIndexName));
 
-            var trainingData = new RoleMappedData(input, roles);
-            var validData = _context == null ? null : new RoleMappedData(_context.ValidationSet, roles);
-
+            var trainingData = new RoleMappedData(trainData, roles);
+            var validData = validationData == null ? null : new RoleMappedData(validationData, roles);
             using (var ch = Host.Start("Training"))
-            using (var pch = Host.StartProgressChannel("Training"))
             {
                 model = TrainCore(ch, trainingData, validData);
             }
 
-            return new MatrixFactorizationPredictionTransformer(Host, model, input.Schema, MatrixColumnIndexName, MatrixRowIndexName);
+            return new MatrixFactorizationPredictionTransformer(Host, model, trainData.Schema, MatrixColumnIndexName, MatrixRowIndexName);
         }
 
+        /// <summary>
+        /// Train a matrix factorization model based on the input <see cref="IDataView"/> using the roles specified by XColumn and YColumn in <see cref="MatrixFactorizationTrainer"/>.
```

Comment:
XColumn and YColumn no longer exist. Please use MatrixColumnIndexKind and MatrixRowIndexKind.